Engine/ac/dynobj/cc_dynamicarray.cpp

```diff
@@ -189,6 +189,28 @@ DynObjectRef DynamicArrayHelpers::CreateStringArray(std::vector<ScriptString::Bu
     return arr;
 }
 
+DynObjectRef DynamicArrayHelpers::CreateScriptArray(const std::vector<DynObjectRef> &items, const char* type_name)
+{
+    String type_str = type_name;
+    auto type = std::find_if(ccInstance::GetRTTI()->GetTypes().begin(), ccInstance::GetRTTI()->GetTypes().end(),
+                             [type_str](const RTTI::Type& i) -> bool {return type_str.Equals(i.name); });
+
+    uint32_t size = type->size == 0 ? sizeof(int32_t) : type->size;
```

Comment:
This is a wrong size calculation. If you are creating array of pointers, that's always int32_t, regardless of whether type has size >0 or not. On another hand, size == 0 does not indicate a pointer storage.

Reply:
I adjusted it here https://github.com/adventuregamestudio/ags/pull/2401/commits/82488f296cb4e62b12933d7a82615fde1e0220d4 , I think dynamically in the api we will always be creating a dynamic array of pointers?